docs/template_plugin/tests/functional/op_reference/base_reference_test.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "base_reference_test.hpp"
+
+#include <gtest/gtest.h>
+
+#include "transformations/utils/utils.hpp"
+
+using namespace InferenceEngine;
+
+CommonReferenceTest::CommonReferenceTest(): targetDevice("TEMPLATE") {
+    core = PluginCache::get().ie(targetDevice);
```

Comment:
can we register template plugin dynamically? And remove `ENABLE_TEMPLATE` option from cmake

Reply:
Example is [here](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/plugin/conformance/test_runner/src/core_config.cpp)

Reply:
Fixed